wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -133,29 +133,76 @@ BeatmapSetID (Integer) is the ID of the beatmap set.
 Difficulty
 ----------
 
-HPDrainRate (Decimal) specifies the HP Drain difficulty.
+The following 4 difficulty properties are expressed in *stars*, where 0 star means very easy, and 10 stars means very hard.
 
-`HPDrainRate:5`
+```
+HPDrainRate:4
+CircleSize:3.3
+OverallDifficulty:4
+ApproachRate:4.8
+```
+
+*HPDrainRate* specifies how fast the health bar decreases. At the time of this writing, the near future definition of this property remains unsettled.
+
+### Circle Size
+
+*CircleSize* (CS) defines the size of the hit circles and sliders in the osu!standard mode.
 
-CircleSize (Decimal) specifies the size of hit object circles.
+The resulting circle radius in osu!pixels is defined by the formula `32 * (1 - 0.7 * (CircleSize - 5) / 5)`, alternatively written `54.4 - 4.48 * CircleSize`.
 
-`CircleSize:4`
+Ranked beatmaps must have a CircleSize value between 2 and 7, inclusive.
 
-OverallDifficulty (Decimal) specifies the amount of time allowed to click a hit object on time.
+In osu!mania mode, *CircleSize* is the number of columns.
 
-`OverallDifficulty:6`
+### Overall Difficulty
 
-ApproachRate (Decimal) specifies the amount of time taken for the approach circle and hit object to appear.
+*OverallDifficulty* (OD) is the harshness of the hit window and the difficulty of spinners.
```

Comment:
harshness into size/stretch/width? not sure

Reply:
I've used *width* in the table that follows it, but I like harshness. It might not be the most accurate word, but it's easy to picture as a human being :)